core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java

```diff
@@ -254,6 +311,14 @@ private static String generateMeterTag(String parentTag, Set<HttpMethod> methods
         addMediaTypes(name, "consumes", consumes);
         addMediaTypes(name, "produces", produces);
 
+        if (!paramPredicates.isEmpty()) {
+            name.add("params");
+            name.add(meterTagJoiner.join(paramPredicates.stream().map(RoutingPredicate::name).iterator()));
+        }
+        if (!headerPredicates.isEmpty()) {
+            name.add("headers");
+            name.add(meterTagJoiner.join(headerPredicates.stream().map(RoutingPredicate::name).iterator()));
```

Comment:
Maybe `distinct()`, just in case there are more than one predicate for the same header name?